rabbitmq_pika_flask/RabbitMQ.py

```diff
@@ -110,40 +111,50 @@ def init_app(
                 parse received messages. Defaults to None.
             msg_parser (Callable, optional): A parser function to
                 parse messages to be sent. Defaults to None.
-            development (bool, optional): If the app is in development mode. Defaults to False.
+            development (bool, optional): Overrides development mode checks. Defaults to False.
             on_message_error_callback (Callable, optional): Function that's called when the processing of
                 a message fails due to an exception.
             middlewares: List of callables that are called, in order, to process a rabbitmq
                 message received from the queue, before finally calling the user consumer func.
         """
 
         self.app = app
-        self.queue_prefix = queue_prefix
         self.config = app.config
 
-        self._check_env()
-
+        self.queue_prefix = queue_prefix
         self.body_parser = body_parser
         self.msg_parser = msg_parser
-        self.development = development
-        self.exchange_name = self.config["MQ_EXCHANGE"]
         self.on_message_error_callback = on_message_error_callback
         self.middlewares.extend(middlewares or [])
 
-        params = URLParameters(self.config["MQ_URL"])
-        self.get_connection = lambda: BlockingConnection(params)
+        self.exchange_name = self.config.get("MQ_EXCHANGE") or os.getenv("MQ_EXCHANGE")
+        assert (
+            self.exchange_name,
+            "MQ_EXCHANGE not set. Please define a default exchange name.",
+        )
+        mq_url = self.config.get("MQ_URL") or os.getenv("MQ_URL")
+        assert (
+            mq_url,
+            "MQ_URL not set. Please define the RabbitMQ url using this format: https://pika.readthedocs.io/en/stable/examples/using_urlparameters.html",
+        )
+        self.development = (
+            development
+            or self.config.get("ENV") == "development"
+            or os.getenv("FLASK_ENV") == "development"
+            or self.config.get("DEBUG")
+            or os.getenv("FLASK_DEBUG")
+        )
```

Comment:
I don't think this works as expected.

`os.getenv` always returns a `str|None`, so if I pass `FLASK_DEBUG=0`, it will still count as a truthy value. Maybe `os.getenv("FLASK_DEBUG") == "1"`. I think `app.config.get` is similar.

Also, if I decide to override the environment variables by passing `development` in, I will only be able to override from False to True, but not vice-versa. Because if `development is False`, it will simply check the variables to find the first expression that's truthy. I believe it would be better to make `development: Optional[bool] = None` in the method args, and then do: `self.development = development if development is not None else ( check env vars here )`.

```suggestion
        self.development = development if development is not None else (
            self.config.get("ENV") == "development"
            or os.getenv("FLASK_ENV") == "development"
            or self.config.get("DEBUG") == "1"
            or os.getenv("FLASK_DEBUG") == "1"
        )
```

Reply:
Just one last change request, I promise :)!